src/dodal/devices/eiger.py

```diff
@@ -79,8 +82,17 @@ def async_stage(self):
         status_ok, error_message = self.odin.check_odin_initialised()
         if not status_ok:
             raise Exception(f"Odin not initialised: {error_message}")
+        self.arming_status = self.do_arming_chain()
+        return self.arming_status
 
-        return self.do_arming_chain()
+    def stage(self):
+        if self.arming_status.done is False:
+            # Arming has started so wait for it to finish
+            self.arming_status.wait(60)
+        else:
+            if self.odin.fan.ready.get() != 1:
+                # Arming hasn't started, do it asynchronously
+                self.async_stage()
```

Comment:
Must: I think in this case we need to be synchronous so we should have a `wait` here